src/models/tag/stream.rs

```diff
@@ -90,133 +116,157 @@ impl FromIterator<HotTag> for HotTags {
-            if !hot_tags_score.is_empty() && !hot_tags_users.is_empty() {
-                Self::put_index_sorted_set(&TAG_GLOBAL_HOT, hot_tags_score.as_slice(), None, None)
-                    .await?;
-                // Add all the users_id in the SET
-                for (label, user_list) in hot_tags_users.into_iter() {
-                    let values_ref: Vec<&str> = user_list.iter().map(|id| id.as_str()).collect();
-                    Taggers::put_index_set(&[label], &values_ref, None, None).await?;
-                }
-            }
+    fn get_cache_key_parts(tags_query: &HotTagsInput) -> Vec<String> {
+        match &tags_query.tagged_type {
+            Some(tagged) => vec![tags_query.timeframe.to_string(), tagged.to_string()],
+            None => vec![tags_query.timeframe.to_string(), String::from("All")],
         }
-        Ok(())
     }
 
-    pub async fn get_global_tags_stream(
-        skip: Option<usize>,
-        limit: Option<usize>,
-        taggers_limit: Option<usize>,
-    ) -> Result<Option<Self>, DynError> {
-        let hot_tags = Self::try_from_index_sorted_set(
-            &TAG_GLOBAL_HOT,
+    async fn get_from_global_cache(tags_query: &HotTagsInput) -> Result<Option<HotTags>, DynError> {
+        let key_parts = HotTags::get_cache_key_parts(tags_query);
+        let key_parts_vector: Vec<&str> =
+            key_parts.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
+        let data = HotTagsData::try_from_index_json(key_parts_vector.clone().as_slice()).await?;
+        let ranking = HotTags::try_from_index_sorted_set(
+            key_parts_vector.as_slice(),
             None,
             None,
-            skip,
-            limit,
+            Some(tags_query.skip),
+            Some(tags_query.limit),
             SortOrder::Descending,
-            None,
+            Some(GLOBAL_HOT_TAGS_PREFIX),
         )
-        .await?
-        .unwrap_or_default();
+        .await?;
 
-        if hot_tags.is_empty() {
+        if data.is_none() || ranking.is_none() {
             return Ok(None);
         }
 
-        // Collect the labels as a vector of string slices
-        let labels: Vec<&str> = hot_tags.iter().map(|(label, _)| label.as_str()).collect();
-        let label_slice: &[&str] = &labels;
-
-        let list = Taggers::try_from_multiple_sets(None, label_slice, taggers_limit).await?;
-
-        let hot_tags_stream: HotTags = hot_tags
-            .into_iter()
-            .zip(list)
-            .filter_map(|((label, score), user_ids)| match user_ids {
-                Some((tagger_list, taggers_count)) => {
-                    let taggers_id = Taggers::from_vec(tagger_list);
-                    Some(HotTag {
-                        label,
-                        taggers_id,
-                        post_count: score as u64,
-                        taggers_count,
-                    })
-                }
-                None => None,
-            })
+        let mapping = data.unwrap();
```

Comment:
:smiling_face_with_tear: Prompt: _I cry when `unwrap` is used. I don't want to cry. Please fix so I don' cry._